odcockpit/include/plugins/environmentviewer/EnvironmentViewerGLWidget.h

```diff
@@ -140,13 +147,26 @@ class SelectableNodeDescriptor;
                     std::shared_ptr<odcore::wrapper::SharedMemory> m_velodyneSharedMemory;
                     bool m_hasAttachedToSharedImageMemory;
                     odcore::data::SharedPointCloud m_velodyneFrame;
-                    const float START_V_ANGLE = -15.0;//For each azimuth there are 16 points with unique vertical angles from -15 to 15 degrees
-                    const float V_INCREMENT = 2.0;  //The vertical angle increment for the 16 points with the same azimuth is 2 degrees
-                    bool m_CPCreceived;//Set to true when the first compact point cloud is received
+                    const float START_V_ANGLE = -15.0; //For each azimuth there are 16 points with unique vertical angles from -15 to 15 degrees
+                    const float V_INCREMENT = 2.0; //The vertical angle increment for the 16 points with the same azimuth is 2 degrees
+                    const float START_V_ANGLE_32 = -30.67; //The starting angle for HDL-32E. Vertical angle ranges from -30.67 to 10.67 degress, with alternating increment 1.33 and 1.34
+                    const float V_INCREMENT_32_A = 1.33; //The first vertical angle increment for HDL-32E
+                    const float V_INCREMENT_32_B = 1.34; //The second vertical angle increment for HDL-32E
+                    uint8_t m_12_startingSensorID_32; //From which layer for the first part(12 layers) of CPC for HDL-32E
+                    uint8_t m_11_startingSensorID_32; //From which layer for the second part(11 layers) of CPC for HDL-32E
+                    uint8_t m_9_startingSensorID_32; //From which layer for the third part(9 layers) of CPC for HDL-32E
+                    float m_12_verticalAngles[12]; //Store the 12 vertical angles for the first part (including 12 layers) of CPC for HDL-32E
```

Comment:
Please use std::array over the old C-style arrays as they are type-safe and provide generic STL-container-like access: http://en.cppreference.com/w/cpp/container/array